Crucible/SchemaCore/SchemaTranslators/ClassTranslator.cs

```diff
@@ -0,0 +1,227 @@
+    /// <inheritdoc/>
+    public string GetEquivalentType(string cSharpType) => Conversions.GetEquivalentJsonType(cSharpType);
+
+    /// <summary>
+    /// Inserts <paramref name="newValue"/> into <paramref name="collection"/>
+    /// with string designation <paramref name="valueName"/> and returns a
+    /// new instance of the class object with the value inserted.
+    /// </summary>
+    /// <exception cref="ArgumentException">If <paramref name="collection"/>'s
+    /// base type does not contain a property named <paramref name="valueName"/>
+    /// or if it does contain a property of that name, but the type is not
+    /// compatible with <typeparamref name="TDefaultValueType"/></exception>
+    /// <typeparam name="TDefaultValueType">Type of the value to be inserted.</typeparam>
+    /// <param name="collection">The collection from which to extract the value.</param>
+    /// <param name="valueName">The string designator for the value to be extracted.</param>
+    /// <param name="newValue">The new value to be inserted.</param>
+    /// <returns>New object with value inserted.</returns>
+    public object InsertFieldValue<TDefaultValueType>(object collection, string valueName, TDefaultValueType newValue)
+    {
+      try
+      {
+        PropertyInfo valueProperty = collection.GetType().GetProperty(valueName);
+        if(valueProperty.PropertyType == typeof(TDefaultValueType))
+        {
+          valueProperty.SetValue(collection, newValue);
+        }
+        else
+        {
+          valueProperty.SetValue(collection, Convert.ChangeType(newValue, valueProperty.PropertyType));
+        }
+        return collection;
+      }
+      catch
+      {
+        throw new ArgumentException($"Attempted to set property {valueName} on class type {collection.GetType().FullName} to value with type {typeof(TDefaultValueType).FullName}");
+      }
+    }
+    /// <inheritdoc/>
+    public bool FieldValueIsNullOrEmpty(object collection, string valueName)
+    {
+      PropertyInfo valueProperty = collection.GetType().GetProperty(valueName);
+      Type valueType = valueProperty.PropertyType;
+      object value = valueProperty.GetValue(collection);
+      return value != null
+        && (valueType.IsAssignableFrom(typeof(JToken)) ? ((JToken)value).IsNullOrEmpty() : value.ToString().IsNullOrEmpty());
+    }
+    /// <inheritdoc/>
+    public bool TryCastValue<TCastType>(object collection, string valueName, out TCastType outputValue)
+    {
+      PropertyInfo valueProperty = collection.GetType().GetProperty(valueName);
+      object value = valueProperty.GetValue(collection);
+      Type valueType = valueProperty.PropertyType;
+      Type castType = typeof(TCastType);
+
+      return InnerTryCastValue(value, valueType, castType, out outputValue);
+    }
+
+    private bool InnerTryCastValue<TCastType>(object value, Type valueType, Type castType, out TCastType outputValue)
```

Comment:
You have `TCastType` and `castType` when you only need one. Is there another purpose for this?

Also document this method.